Content.Server/Power/Generator/ChemicalFuelGeneratorAdapterComponent.cs

```diff
@@ -11,14 +12,29 @@ namespace Content.Server.Power.Generator;
 public sealed class ChemicalFuelGeneratorAdapterComponent : Component
 {
     /// <summary>
-    /// The acceptable list of input entities.
+    /// The reagent to accept as fuel.
     /// </summary>
-    [DataField("whitelist")]
-    public EntityWhitelist? Whitelist;
+    [DataField("reagent", customTypeSerializer: typeof(PrototypeIdSerializer<ReagentPrototype>))]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public string Reagent = "WeldingFuel";
 
     /// <summary>
-    /// The conversion factor for different chems you can put in.
+    /// The solution on the <see cref="SolutionContainerManagerComponent"/> to use.
     /// </summary>
-    [DataField("chemConversionFactors", required: true, customTypeSerializer:typeof(PrototypeIdDictionarySerializer<float, ReagentPrototype>))]
-    public Dictionary<string, float> ChemConversionFactors = default!;
```

Comment:
any reason the `ChemConversionFactors` was replaced with just a single reagent datafield?

Reply:
Because I couldn't think of a good way to use it with the rewritten logic:
* Right now the fuel values in the UI are 1:1 with the reagent content, which I think is good.
* I'd need to make it so the fuel consumption system picks from all reagents in the solution at some ratios, change the burn time depending on reagent, etc...

Reply:
aight